bazel/thirdparty/seastar.BUILD

```diff
@@ -547,6 +583,15 @@ cc_library(
         # build settings. defining for all compilation units seems harmless.
         ":use_heap_profiling": ["SEASTAR_HEAPPROF"],
         "//conditions:default": [],
+    }) + select({
+        ":use_system_allocator": ["SEASTAR_DEFAULT_ALLOCATOR"],
+        "//conditions:default": [],
+    }) + select({
+        ":with_debug": ["SEASTAR_DEBUG"],
+        "//conditions:default": [],
+    }) + select({
+        ":with_shuffle_task_queue": ["SEASTAR_SHUFFLE_TASK_QUEUE"],
```

Comment:
I think there is also an option for guard pages for ss::thread

Reply:
I suppose it might be good to talk about when to enable some of these seastar flags

Reply:
> I think there is also an option for guard pages for ss::thread

added this option.

> I suppose it might be good to talk about when to enable some of these seastar flags

there is a todo item on the wiki for this. basically at some point we need to take full stock of all the dependencies and their available flags and figure out all the different build configurations we want. i've mostly tried to use 3rdparty.cmake as a guide here, but for custom builds like seastar, it's also necessary to go in and make sure all the defaults are replicated as well.